src/core/sorted_map.cc

```diff
@@ -769,5 +769,16 @@ SortedMap* SortedMap::FromListPack(PMR_NS::memory_resource* res, const uint8_t*
   return zs;
 }
 
+bool SortedMap::DefragIfNeeded(float ratio) {
+  bool realloced = false;
+
+  for (auto it = score_map->begin(); it != score_map->end(); ++it) {
+    realloced |= it.ReallocIfNeeded(ratio);
+    // Not sure if it's worth going over B+ tree cause lgn to delete + lgn to insert
```

Comment:
Yeah, I don't think it's really worth going over the B+ tree because we would need to delete + insert. We can add this in the future if we wish to + it's really boilerplate....

Reply:
+ @romange here as well
sorry, I'm not familiar with this class